docs/rules/no-incorrect-deep-equal.md

```diff
@@ -0,0 +1,28 @@
+# Avoid using `deepEqual` with literals
+
+`deepEqual` and `notDeepEqual` are unnecessary when comparing literals, template strings or `undefined`. Use `is` or `not` instead.
```

Comment:
Primitives is a more correct word than literals (there could be object literal).

```suggestion
The `deepEqual` and `notDeepEqual` assertions are unnecessary when comparing primitives. Use `is` or `not` instead.
```